src/solver/conv_hip_implicit_gemm_mlir_cpp_fwd.cpp

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2019 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/conv/invokers/impl_gemm.hpp>
+#include <miopen/solver.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/generic_search.hpp>
+
+#include "implicitgemm_util.hpp"
+
+namespace miopen {
+namespace solver {
+
+bool ConvHipImplicitGemmMlirCppFwd::IsApplicable(const ConvolutionContext& ctx) const
+{
+#ifdef MIOPEN_LIBMLIRMIOPEN
+    if(ctx.Is3d())
+        return false;
+    return ConvHipImplicitGemmV4R4Fwd::IsApplicable(ctx);
+#else
+    std::ignore = ctx;
+    return false;
+#endif
+}
+
+PerformanceImplicitGemmMlirCppFwd
+ConvHipImplicitGemmMlirCppFwd::GetPerformanceConfig(const ConvolutionContext& ctx) const
+{
+    return GetPerformanceConfigBase<PerformanceImplicitGemmMlirCppFwd>(ctx);
+}
+
+PerformanceImplicitGemmMlirCppFwd
+ConvHipImplicitGemmMlirCppFwd::Search(const ConvolutionContext& context,
+                                      const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, context, invoke_ctx);
+}
+
+ConvSolution ConvHipImplicitGemmMlirCppFwd::GetSolution(
+    const ConvolutionContext& ctx, const PerformanceImplicitGemmMlirCppFwd& config, bool) const
+{
+    ConvSolution result;
+    KernelInfo construction_parameters;
+
+    assert(config.IsValid(ctx));
```

Comment:
Do we expect this be always valid in Release ?

Reply:
I was assuming release build will get rid of asserts. Is it not the case?

Reply:
I double checked other solvers, in particular `conv_hip_implicit_gemm_fwd_v4r4.cpp`. It is following the same use convention as in there. If assert is not acceptable, we should file an issue and completely remove all asserts from the code base. Marking it as resolved for now.